packages/integration-template-tests/lib/sync-template.ejs

```diff
@@ -0,0 +1,116 @@
+import { vi, expect, it, describe } from "vitest";
```

Comment:
Going back to the idea of having a recording adapter for proxy, batch, etc... to have all the http mocking logic in one place instead of having to duplicate it in the tests. By this I mean something like 
```
Recorder.proxy({proxyConfig: config, option: this.recordingOption) 
// with recordingOption of type = 'record' | 'replay' | 'passthrough'
```
when dry-run --save-responses is set, set 'record' as the recordingOption. The proxy function then make the real http request and save the content in a file. if --save-responses is not set, the recordingOption is 'passthrough', the default value. In thi case `Recorder.proxy` make the real http request and don't save anything

In the tests just set NangoProps.recordingOptions to 'replay', in this case `Recorder.proxy()` get the response from the file.

Same for `batchSave` or any other operations requiring mocking. In batchSave for example, if `this.recordingOptions` is `record`, save the output in a file and when set to `replay` get the data from the file. 

It would mean:
- adding a single new property `recordingOption` to NangoProps/NangoAction
- changes to sync/sdk will be minimal. Basically in the functions that requires mocking replace `await this.nango.proxy(proxyConfig)` into `Recorder.proxy({proxyConfig, nango: this}) or add `Recorder.batchSave({results, nango: this})`, etc...
- no need for mocking anything in the test, just setting the nangoMock.recordingOption to `replay` and calling `await fetchData(nangoMock);`
- all the file manipulation logic (reading and writing) is in the Recorder

What do you think? I am happy to pair if you prefer

Reply:
I like the idea but unsure I'll have the bandwidth to make the update this week. I follow the high level but would want to discuss the finer details probably via pair but do you feel this is an immediate blocking update or a v2 change?

Reply:
Discussed this with Thomas and I am going to revisit the approach next week to potentially mock at the axios level to simplify things further. However will merge this and make that improvement on the next iteration.